WordPress/Classes/Services/AccountService.m

```diff
@@ -287,4 +287,24 @@ - (void)updateAccount:(WPAccount *)account withUserDetails:(RemoteUser *)userDet
     [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
 }
 
+- (void)purgeSelfHostedAccount:(WPAccount *)account
+{
+    if (!account.isWpcom && [account.blogs count] == 0) {
+        [self.managedObjectContext deleteObject:account];
+    }
+}
+
+- (void)purgeJetpackAccount:(WPAccount *)account
```

Comment:
I don't think it's a pre-condition, I think it just indicates intention. If you passed a self hosted account, it'd behave similarly to `purgeSelfHostedAccount:`

This could be called `purgeDotComAccountIfItsNotBeingUsedAnymore:`, but this name hints that it's meant to be used to clean up Jetpack accounts